Fix admin permissions hover

ceolson01 opened the pull request:
## Description

This PR adds a tooltip (er... `Popover`, to be technical) when a user hovers over a disabled permission in the colony permissions edit dialog.

**New stuff** ✨

* `PermissionCheckbox` component

**Changes** 🏗

* Use `PermissionCheckbox` component in `ColonyPermissionsEditDialog`
* Improve clarity of some logic regarding state updates
* Use global i18n messages for buttons and some roles.

**Demo** 🎥 

![Permissions Hover](https://user-images.githubusercontent.com/3052635/65792609-bd754a00-e129-11e9-90b3-4ed2cfc55fc2.gif)


Closes #1852

chmanie commented:
I should base my 1850 on this

sprusr commented on src/modules/admin/components/Permissions/ColonyPermissionEditDialog.tsx:
```diff
@@ -267,9 +236,9 @@ const ColonyPermissionEditDialog = ({
-    const { data } = useUserDomainRoles(colonyAddress, domain.id, selectedUser);
+  const { data } = useUserDomainRoles(colonyAddress, domain.id, selectedUser);
 
+  useEffect(() => {
```
Nice improvement, definitely wanted doing 👍

sprusr commented on src/modules/admin/components/Permissions/ColonyPermissionEditDialog.tsx:
```diff
@@ -340,34 +307,25 @@ const ColonyPermissionEditDialog = ({
-                    </span>
-                  </Checkbox>
+                    role={role}
+                    roleDescription={MSG[`roleDescription${idx}`]}
```
How come move the message handling for the role name to `PermissionCheckbox` but not for the description?

ceolson01 replied:
So the role names are actually in the `en/i18n.json` file, but `PermissionCheckbox` does indeed have the _logic_ to piece together the message id strings.

That said, perhaps it would make sense to move the description messages into `PermissionCheckbox` and use the role name to access each message... I think I'll do that

ceolson01 commented:
@sprusr I moved the role description messages into the `PermissionCheckbox` component, and also added styles to use the `pointer` cursor for checkboxes.

Before I merge this, could you please check out those last couple of changes?

sprusr commented on src/modules/admin/components/Permissions/PermissionCheckbox.tsx:
```diff
@@ -33,9 +60,11 @@ const PermissionCheckbox = ({
-  roleDescription,
 }: Props) => {
   const roleNameMessage = { id: `role.${role.toLowerCase()}` };
+  const roleDescriptionMessage = MSG[
```
Could this be inside the `useMemo`?

ceolson01 replied:
Yessir! Good catch, thank you 🙏
